pkg/database/postgres.go

```diff
@@ -38,8 +38,19 @@ func NewPGDatabase(conf *entities.Config) *PGDatabase {
 		dbInstance = &PGDatabase{DB: bun.NewDB(pgDB, pgdialect.New())}
 		fmt.Printf("%+v", dbInstance)
 		ctx := context.Background()
+		// Create the user table
+		createTableSQL := `
+		CREATE TABLE IF NOT EXISTS public.user (
+			id SERIAL PRIMARY KEY,
+			name VARCHAR(255) NOT NULL,
+			username VARCHAR(255) UNIQUE NOT NULL
+		);`
+		_, err = dbInstance.DB.ExecContext(ctx, createTableSQL)
+		if err != nil {
+			log.Fatalf("Failed to create table: %v", err)
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider graceful error handling for DB</b></div>
<div id="fix">

Consider handling table creation failure more gracefully instead of using `log.Fatalf`. This allows for proper cleanup and error propagation.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
 		_, err = dbInstance.DB.ExecContext(ctx, createTableSQL)
 		if err != nil {
			return fmt.Errorf("failed to create table: %w", err)
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/15#issuecomment-2563481808>#d7fde1</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged